src/sources/http.ts

```diff
@@ -47,14 +86,26 @@ export const createHTTPSource: SourceFactory = (options: any) => {
 
     let mtime
     const _lastModified = response.headers.get('last-modified')
+    const _etag = response.headers.get('etag')
+
     if (_lastModified) {
       mtime = new Date(_lastModified)
     }
 
+    let buffer
+
+    if (options.cache) {
+      const filename = join(options.cache, 'data', `${murmurhash(id)}`)
```

Comment:
If we don't care about deterministic filenames we could just use a random temp filename and not use murmurhash